### ISSUE TITLE
AllRowsReader should build tokenRange with given DC or RACK

### ISSUE THREAD

**Issue by cywjackson** [Netflix/astyanax#243]
Currently it just uses 
List<TokenRange> ranges = keyspace.describeRing();

In a case where I have 2 DC (and replicating on both DC),  the AllRowsReader would send the request to both DC, which is not what I really want as I only want it to be run on 1 DC (local to my client).

There should be field for DC and Rack (nullable) in the reader, and the builder would have the method to populate them. The "describeRing()" should be replaced with "describeRing(String dc, String rack)"

Basically anywhere A6x uses describeRing() should be deprecated to use describeRing(final String dc, final String rack)  anyway...

any suggestion on workaround for now?
